src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -547,4 +548,39 @@ private function addPropertyInfoSection(ArrayNodeDefinition $rootNode)
             ->end()
         ;
     }
+
+    private function addCacheSection(ArrayNodeDefinition $rootNode)
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('cache')
+                    ->info('Cache configuration')
+                    ->fixXmlConfig('pool')
+                    ->children()
+                        ->arrayNode('pools')
+                            ->useAttributeAsKey('name')
+                            ->prototype('array')
+                                ->children()
+                                    ->scalarNode('type')
+                                        ->info('The cache pool type (one of "default", "apcu", "doctrine", "psr6", "filesystem" or any other existing cache.adapter.* service name.)')
+                                        ->defaultValue('default')
+                                    ->end()
+                                    ->booleanNode('public')->defaultFalse()->end()
+                                    ->integerNode('default_lifetime')->defaultNull()->end()
+                                    ->scalarNode('provider_service')->defaultNull()->end()
+                                    ->scalarNode('directory')->defaultNull()->end()
+                                ->end()
+                                ->validate()
+                                    ->ifTrue(function ($v) {
+                                        return isset($v['provider_service']) xor in_array($v['type'], array('psr6', 'doctrine'));
```

Comment:
According to the error message it should be 
`return isset($v['provider_service']) xor !in_array($v['type'], array('psr6', 'doctrine'));`